RUFAS/data_validator.py

```
@@ -1870,6 +1870,72 @@ def validate_beef_cow_calf_config(config: dict[str, Any]) -> None:
+                raise ValueError("stocker_exit_weight must exceed stocker_entry_weight")
+
+    @staticmethod
+    def _validate_stocker_diet_system(config: dict[str, Any]) -> None:
+        """Raise ValueError for an unrecognised stocker_diet_system value."""
+        from RUFAS.biophysical.animal.data_types.animal_enums import StockerDietSystem
+
+        if "stocker_diet_system" in config and config["stocker_diet_system"] is not None:
+            system = str(config["stocker_diet_system"])
+            valid = {m.value for m in StockerDietSystem}
+            if system not in valid:
+                raise ValueError(f"stocker_diet_system must be one of {sorted(valid)}, got '{system}'")
```

Ruff (0.15.21)

[warning] Avoid specifying long messages outside the exception class

(TRY003)